### ISSUE TITLE
More recent data and covariates

### ISSUE THREAD

**Issue by DavidTallmon** [mdscheuerell/AukeCoho#5]
Hey guys. Apologies for being AWOL.  I'm back to working on this with some updated data and covariates.  The data are complete through 2017 but lack age structure information from scale aging for 2018-2019.  Not sure if such data are beneficial.  The covariates are ones shown to be somewhat informative from recent LME work.  These covariates overlap with the last set we examined but have some differences.  
[covariates_1980-2019.xlsx](https://github.com/mdscheuerell/AukeCoho/files/4788402/covariates_1980-2019.xlsx)

[auke_coho_data_1980-2019.xlsx](https://github.com/mdscheuerell/AukeCoho/files/4788401/auke_coho_data_1980-2019.xlsx)

**Comment by ebuhle** [comment]
Hi @DavidTallmon, welcome to the AWOL club. Great to hear you've got some new data. The lack of recent age-frequency data isn't a problem, although of course it will be reflected in the posterior uncertainty -- always beneficial, but not necessary.

**Comment by ebuhle** [comment]
<img src=https://user-images.githubusercontent.com/22503560/89845193-a87f8a80-db32-11ea-8cb7-906f9fd9fc7d.png>

So I'm getting reacquainted with the environmental covariate data. We have three variables:
* `hpc_release`
* `pdo_nov_jan`
* `gauge_spring`

IIRC, the hatchery releases (of what species again? sorry) is a predictor of SAR for smolts entering the ocean in the same year; winter PDO is a predictor of SAR for smolts entering the ocean the following spring (i.e., the same year as the Jan value in the 3-mo average); and I assume spring discharge is also a predictor of SAR for smolts outmigrating in the same year. Do I have that right? I particularly want to make sure the Nov-Jan PDO values are indexed to the "Jan" year, not the "Nov" year.

I'm also remembering that the way `IPM_SMaS_np` is set up now, the same regression coefficients apply to SAR regardless of smolt age. That seems reasonable on its face, but there is that startling lack of correlation b/w SAR for 2- vs. 3-year-old smolts suggesting that they're doing something different at sea. Thoughts?

Last question for now: do we not have any predictors of spawner-to-smolt productivity?

**Comment by DavidTallmon** [comment]
I guess you can't get out alive.  
We had thought of some of this the same.  HPC is # hatchery releases of pink and chum and we had envisioned releases to increase first summer ocean survival by serving as a predation buffer.  We had thought of high spring discharge as decreasing spawner-to-smolt productivity - via scour and early downstream displacement of juvies.  And, we had thought of PDO as affecting first ocean winter productivity and food availability -- so positively affecting only 1 SW.  It is currently indexed to the Dec-Nov year, NOT Jan year.  But, I think the way you described PDO effects are plausible.
I can't come up with a good explanation of the 2-, 3-yr old smolt SAR differences off the top of my head. I'll think about it and pass it by the guys at the weir.

**Comment by ebuhle** [comment]
> We had thought of some of this the same. HPC is # hatchery releases of pink and chum and we had envisioned releases to increase first summer ocean survival by serving as a predation buffer. 

Oh, `h`atchery `p`ink `c`hum, duh.

> We had thought of high spring discharge as decreasing spawner-to-smolt productivity - via scour and early downstream displacement of juvies. 

Oh, I see. So is this the spring freshet? I can see downstream displacement for sure, but are there still eggs in the gravel at that point? Or alevins? I guess things happen later up there.

> And, we had thought of PDO as affecting first ocean winter productivity and food availability -- so positively affecting only 1 SW. It is currently indexed to the Dec-Nov year, NOT Jan year. But, I think the way you described PDO effects are plausible.

Ah, OK, that makes sense. Guess I'm used to thinking of winter PDO as "preconditioning" in the Logerwell sense. But first-winter survival makes sense too. And I always forget that warm PDO up there is good, yes?

I had originally written a caveat about factors affecting first-winter survival that the model will apply equally to jacks, but then I realized I was confused about my own parameterization. SAR and conditional age-at-maturity are both functions of the underlying survival and maturation schedule; SAR is a weighted average of jacks and 1SW. With that in mind, it's fine.

**Comment by ebuhle** [comment]
> I can't come up with a good explanation of the 2-, 3-yr old smolt SAR differences off the top of my head. I'll think about it and pass it by the guys at the weir.

Don't think it hadn't crossed my mind that this is exactly what you'd expect to see if FW age were _not_ sampled randomly conditional on SW age.

**Comment by ebuhle** [comment]
> > We had thought of high spring discharge as decreasing spawner-to-smolt productivity - via scour and early downstream displacement of juvies.
> 
> Oh, I see. So is this the spring freshet? I can see downstream displacement for sure, but are there still eggs in the gravel at that point? Or alevins? I guess things happen later up there.

Oh wait, so this should be indexed to the brood year, not the calendar year of the gage measurement.

**Comment by DavidTallmon** [comment]
Yes to all of the above.  Generally, warm/positive PDO is good for stocks in northern SEAK....or at least data have suggested that's the case historically.  Litzow et al question whether PDO is a useful predictor of ocean survival for AK stocks in last few yrs.  WRT the spring gauge we suspect spring freshets push alevins/new parr downstream before they're ready...maybe more often in recent springs.  We don't get ice out on the lake until late April, but its happening earlier.  We're definitely slow up here, relative to WA.  It should be indexed to brood year, not calendar year of the measurement.  
Agreed, the whole 2-, 3-yr SAR disparity is a bit unsettling.

**Comment by ebuhle** [comment]
>  It should be indexed to brood year, not calendar year of the measurement.

Sorry, just being thick: that's the way it is in the dataset now?

**Comment by ebuhle** [comment]
> I can't come up with a good explanation of the 2-, 3-yr old smolt SAR differences off the top of my head. I'll think about it and pass it by the guys at the weir.

Here's a thought: age-2 smolts are (apparently) 30% more likely to return as 1SW (97%) than age-3 smolts are (75%). If there are strong drivers that affect first-winter survival, like say PDO, they would show up more in age-2 SAR. They'd have to be pretty strong to account for that huge bump in the early '00s. Guess we'll find out by fitting them.

**Comment by DavidTallmon** [comment]
At present, the flow is presented in the year measured, so it would need to be adjusted by script to index the brood year it affects.  
I'm still trying to make peace with the age specific SAR patterns.  There's definitely historic evidence of PDO affecting first winter survival....not sure if its THAT strong.  There's also historic evidence of detectable effects of first few months at sea on age-2 and age-3 outmigrants in the same year.....that is, analyses of the first ~20 years of data showed 0SW returns were decent predictors 1SW returns the next fall.

**Comment by ebuhle** [comment]
> At present, the flow is presented in the year measured, so it would need to be adjusted by script to index the brood year it affects.

OK, but then the value for BY 2019 is missing. Is the 2020 gage reading available yet?

> I'm still trying to make peace with the age specific SAR patterns. There's definitely historic evidence of PDO affecting first winter survival....not sure if its THAT strong. There's also historic evidence of detectable effects of first few months at sea on age-2 and age-3 outmigrants in the same year.....that is, analyses of the first ~20 years of data showed 0SW returns were decent predictors 1SW returns the next fall.

I suppose both things can be true; the sibling relationship could be dominated by smolt cohort strength. I'm still trying to parse the math of this. So far it looks like there's a ~97% probability the winter PDO effect is positive; HPC is a wash. I need to make some plots to explore the relationships more.

**Comment by DavidTallmon** [comment]
I've attached updated covariates file w/ spring 2020 value.
[covariates_1980-2020.xlsx](https://github.com/mdscheuerell/AukeCoho/files/5059656/covariates_1980-2020.xlsx)

Josh said he'd look into historic scale sampling details for insights into age-2 age-3 SARs.

**Comment by ebuhle** [comment]
> I've attached updated covariates file w/ spring 2020 value.
> [covariates_1980-2020.xlsx](https://github.com/mdscheuerell/AukeCoho/files/5059656/covariates_1980-2020.xlsx)

Sweet, thanks. FWIW, the best way to do this is to use the normal version-control process: either add this new file to the repo or edit the existing `covariates_1980-2019.csv`. (Whether to rename it is debatable, as the properly aligned covariates actually span 1980-2019.)

**Comment by ebuhle** [comment]
Here are some marginal effect plots for the covariates of each life stage in the Ricker model. The plot needs a little more finessing, but the take-home is that even the strongest effect (PDO) does not explain all that much of the variance in the estimated states -- and certainly doesn't account for those three anomalously high age-2 SARs in the early 2000s.

<img src=https://user-images.githubusercontent.com/22503560/90069047-8d358c00-dca6-11ea-8c74-62b1458a41f5.png>

**Comment by DavidTallmon** [comment]
Those are pretty figs.  I wish those 3 yrs didn't 'stick out' so much.  I'm not entirely surprised these covars don't explain a ton of the variance in the states, but must admit I had hoped they'd explain more.  I need to get back with Josh about the scale sampling in the earlly 00s.

**Comment by ebuhle** [comment]
It almost looks like the relationship between HPC and age-3 smolt SAR is unimodal. Seems plausible that high hatchery releases might be bad, due to predator aggregation aka apparent competition or (maybe?) actual competition. Not so for age-2, though, apart from those three outliers.

**Comment by DavidTallmon** [comment]
Agreed.  That hump in HPC is intriguing.  I've never been able to completely buy the argument that more HPC is more better.

**Comment by ebuhle** [comment]
Well, should we try a quadratic? I'm leery of fishing expeditions / garden of forking paths, especially since those high-leverage 2000-2002 values would surely drive the relationship for age-2 SAR. But I can honestly say to my inner Burnham & Anderson that I hypothesized _a priori_ the relationship was not monotonically increasing.

**Comment by DavidTallmon** [comment]
I agree. I think we should try a quadratic for HPC.  (As long as the inner B & A remains dormant.)

**Comment by ebuhle** [comment]
Well, there's strong support for the quadratic term (gray line in the density plot) but I don't know if I'd say this really captures what's going on with the states (hence ultimately the data):

<img src=https://user-images.githubusercontent.com/22503560/90344317-99368c00-dfcd-11ea-8bb1-8ea371943b41.png>

As before, the issue is that SAR for 2- vs. 3-year-old smolts appears to be doing completely different things, so shared covariate effects try to fit both and end up not fitting either one very well. (Although I guess the PDO effects are about equally good / bad.)

If / when we are able to do approximate leave-one-out, we'll see whether the addition of the quadratic term actually improves out-of-sample predictive skill.

**Comment by DavidTallmon** [comment]
Hmmm.  Briscoe et al. 2005 suggested that Auke jacks may stay local cause jack return rates have shown a stronger positive relationship to local early summer SST than large returners.  And, while jack return rates have explained some, they did not explain a lot of the variation in large returner survival.  There is some (incomplete) temporal segregation between age 2s and 3s on their way out to sea............maybe conditions then dominate?  Does it seem like we're reaching?
Josh looked back at the scale sampling dates and there was nothing obviously egregious about the sampling in those few years (early 00s) where the age 2s acted like 3s.  
Is the dark line in the HPC plot a linear effect?

**Comment by ebuhle** [comment]
> Hmmm. Briscoe et al. 2005 suggested that Auke jacks may stay local cause jack return rates have shown a stronger positive relationship to local early summer SST than large returners. And, while jack return rates have explained some, they did not explain a lot of the variation in large returner survival. There is some (incomplete) temporal segregation between age 2s and 3s on their way out to sea............maybe conditions then dominate? Does it seem like we're reaching?

Seems reasonable. Chinook and coho jacks in Puget Sound similarly stay close to home. I suppose the ~8x difference in jack proportions and the difference in outmigration timing could explain why age-2 and age-3 smolts have such different outcomes. 

It is reassuring to hear that there's nothing obviously off about the scale sampling in the early '00s. You can clearly see the pattern in the data that drives this. It's that shift among 1SW scales, with a sudden jump in the proportion of `3_2` and corresponding drop in `4_3`. There's a hint of a corresponding shift among jacks 1 year earlier, but it's nowhere near as dramatic or persistent.

<img src=https://user-images.githubusercontent.com/22503560/90451707-41695500-e0a1-11ea-9fc8-0e75f5ee9853.png height="50%" width="50%">

> Is the dark line in the HPC plot a linear effect?

Yeah. It was tricky to find a layout that was both general (vs. hardwired to a particular covariate structure) and clear. The gray density should really be labeled $\beta_{\text{HPC}^2}$ . Also, I used raw polynomial terms (after centering and standardizing `HPC`), so there is some positive posterior correlation between the linear and quadratic coefs. Stan sampled them with aplomb, but the separate marginal densities don't tell the whole story in that sense.
